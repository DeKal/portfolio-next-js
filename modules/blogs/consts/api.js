export const API_END_POINT = 'https://blog-illuminati.herokuapp.com/api/'
