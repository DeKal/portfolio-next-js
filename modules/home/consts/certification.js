export const LEFT_INFO_POS = ['11%', '37%', '62%', '88%']
export const ZINDEX = [40, 30, 20, 10]
export const ITEM = ['c', 'c', 'p', 'p']
export const INFO = [
  {
    title: 'CertBachelor',
    content: 'CertContentBachelor',
    coverMobile: '/images/certs/graduation-mobile.webp',
    cover: '/images/certs/graduation.webp'
  },
  {
    title: 'CertCom',
    content: 'CertContentComm',
    coverMobile: '/images/certs/communication-mobile.webp',
    cover: '/images/certs/communication.webp'
  },
  {
    title: 'CertACMPrize',
    content: 'CertContentACMPrize',
    coverMobile: '/images/certs/award-mobile.webp',
    cover: '/images/certs/award.webp'
  },
  {
    title: 'CertSPrize',
    content: 'CertContentSPrize',
    coverMobile: '/images/certs/award-mobile.webp',
    cover: '/images/certs/award.webp'
  },
  {
    title: 'CertFPrize',
    content: 'CertContentFPrize',
    coverMobile: '/images/certs/award-mobile.webp',
    cover: '/images/certs/award.webp'
  }
]
