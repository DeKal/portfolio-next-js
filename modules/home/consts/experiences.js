export const COMPANIES = [
  { name: 'Zalora', icon: '/images/companies/zalora.png' },
  { name: 'Dek', icon: '/images/companies/dek.png' },
  { name: 'GCS', icon: '/images/companies/gcs.png' },
  { name: 'GCS-intern', icon: '/images/companies/gcs.png' }
]
