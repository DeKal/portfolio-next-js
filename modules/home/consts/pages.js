import React from 'react'
import {
  FaHome,
  FaBriefcase,
  FaAddressBook,
  FaMedal,
  FaCode
} from 'react-icons/fa'

export const HOME = 'Home'
export const ABOUT = 'About'
export const EXPERIENCES = 'Experiences'
export const CERTIFICATIONS = 'Certifications'
export const PROJECTS = 'Projects'
export const CONTACTS = 'Contacts'
export const BLOG = 'Blogs'

export const PAGE_LIST = [
  {
    name: HOME,
    isShowOnMobile: true,
    Icon: <FaHome size="22px" />,
    url: `/#${HOME}`,
    cmpName: HOME
  },
  {
    name: EXPERIENCES,
    isShowOnMobile: true,
    Icon: <FaBriefcase size="22px" />,
    url: `/#${EXPERIENCES}`,
    cmpName: EXPERIENCES
  },
  {
    name: PROJECTS,
    isShowOnMobile: true,
    Icon: <FaCode size="22px" />,
    url: `/#${PROJECTS}`,
    cmpName: PROJECTS
  },
  {
    name: CERTIFICATIONS,
    isShowOnMobile: true,
    Icon: <FaMedal size="22px" />,
    url: `/#${CERTIFICATIONS}`,
    cmpName: CERTIFICATIONS
  },
  {
    name: CONTACTS,
    url: `/#${CONTACTS}`,
    Icon: <FaAddressBook size="22px" />,
    isShowOnMobile: true,
    cmpName: CONTACTS
  },
  {
    name: BLOG,
    isShowOnMobile: false,
    url: `/blogs`,
    cmpName: 'blogs'
  }
]
