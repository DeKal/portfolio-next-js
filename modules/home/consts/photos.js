export const photos = [
  {
    src: '/images/projects/zalora.webp',
    width: 2,
    height: 4,
    title: 'Zalora Lite'
  },
  {
    src: '/images/projects/popswap.webp',
    width: 2,
    height: 4,
    title: 'Popswap'
  },
  {
    src: '/images/projects/bootcareer.webp',
    width: 7,
    height: 4,
    title: 'Bootcareer Jobseeker'
  },
  {
    src: '/images/projects/brand360.webp',
    width: 6,
    height: 5,
    title: 'Brand360 project'
  },
  {
    src: '/images/projects/taman.webp',
    width: 3,
    height: 2,
    title: 'Tam An Web'
  },
  {
    src: '/images/projects/2048.webp',
    width: 6,
    height: 6,
    title: '2048 Fun Games'
  }
]
