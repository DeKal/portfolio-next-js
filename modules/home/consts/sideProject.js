const row1 = ['BootCareer-FE', 'BootCareer-BE']
const row2 = ['Portfolio-FE', 'Portfolio-BE']
const row3 = ['Popswap', 'LiveGuess']

export const getProjects = () => [row1, row2, row3]
