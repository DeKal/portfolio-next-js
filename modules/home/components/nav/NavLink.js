import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { EN } from '~/modules/home/consts/langs'
import { Translate, withLocalize } from 'react-localize-redux'
import useWindowDimensions from '~/modules/core/utils/useWindowDimensions'

const NavLink = ({
  lang,
  selectedLang,
  setSelectedLang,
  setActiveLanguage
}) => {
  const { width } = useWindowDimensions()
  return (
    <Container
      isSelected={selectedLang === lang}
      onClick={() => {
        setSelectedLang(lang)
        setActiveLanguage(lang)
        window.localStorage.setItem('lang', lang)
      }}
    >
      {width < 600 ? (
        lang === EN ? (
          'en'
        ) : (
          'vn'
        )
      ) : lang === EN ? (
        <Translate id="english" />
      ) : (
        <Translate id="vietnamese" />
      )}
    </Container>
  )
}

NavLink.propTypes = {
  lang: PropTypes.string,
  selectedLang: PropTypes.string,
  setSelectedLang: PropTypes.func,
  setActiveLanguage: PropTypes.func
}

export default withLocalize(NavLink)

const Container = styled.a`
  color: ${props =>
    props.isSelected
      ? props.theme.colors.colorDark
      : props.theme.colors.colorBlack};
  cursor: ${props => (props.isSelected ? 'default' : 'pointer')};
  text-decoration: ${props =>
    props.isSelected ? 'underline' : 'none'} !important;
  font-weight: ${props => (props.isSelected ? 'bold' : 'none')};
  border: ${props => (props.isSelected ? 'none' : 'initial')};
  border-color: ${props => props.theme.colors.colorLight};
  font-family: ${props => props.theme.font};
  font-style: italic;
  line-height: ${props => props.theme.navHeaderHeight};
  transition: color 0.5s;

  &:hover {
    color: ${props => props.theme.colors.colorDark};
    border: none;
  }
`
