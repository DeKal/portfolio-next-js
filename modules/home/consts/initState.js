import { HOME } from '~/modules/home/consts/pages'
import { EN } from '~/modules/home/consts/langs'

export const INIT_NAV_CONTENT_SHOW = false
export const INIT_SELECTED_PAGE = HOME
export const INIT_SELECTED_LANG = EN
