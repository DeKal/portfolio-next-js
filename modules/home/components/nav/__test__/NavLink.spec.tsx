import * as React from 'react'
import { mountWithTheme } from '~/modules/core/utils/withThemeProviders'
import NavLink from '~/modules/home/components/nav/NavLink'
import { EN, VN } from '~/modules/home/consts/langs'

import useWindowDimensions from '~/modules/core/utils/useWindowDimensions'
jest.mock('~/modules/core/utils/useWindowDimensions')

describe('Nav Link', () => {
  describe('Big Device >= 768px', () => {
    beforeEach(() => {
      useWindowDimensions.mockReturnValue({
        width: 768
      })
    })

    it('should render with English selected', function() {
      const wrap = mountWithTheme(<NavLink lang={EN} selectedLang={EN} />)
      expect(wrap).toMatchSnapshot()
    })

    it('should render with Vietnamese selected', function() {
      const wrap = mountWithTheme(<NavLink lang={VN} selectedLang={VN} />)
      expect(wrap).toMatchSnapshot()
    })

    it('should change Language when clicking', function() {
      const setSelectedLang = jest.fn()
      const setActiveLanguage = jest.fn()
      const wrap = mountWithTheme(
        <NavLink
          lang={VN}
          selectedLang={EN}
          setSelectedLang={setSelectedLang}
          setActiveLanguage={setActiveLanguage}
        />
      )
      wrap.simulate('click')
      expect(wrap).toMatchSnapshot()

      expect(setSelectedLang.mock.calls.length).toEqual(1)
      expect(setSelectedLang).toHaveBeenCalledWith(VN)

      expect(setActiveLanguage.mock.calls.length).toEqual(1)
      expect(setActiveLanguage).toHaveBeenCalledWith(VN)
    })
  })

  describe('Small Device < 600px', () => {
    beforeEach(() => {
      useWindowDimensions.mockReturnValue({
        width: 500
      })
    })

    it('should render with English selected', function() {
      const wrap = mountWithTheme(<NavLink lang={EN} selectedLang={EN} />)
      expect(wrap).toMatchSnapshot()
    })

    it('should render with Vietnamese selected', function() {
      const wrap = mountWithTheme(<NavLink lang={VN} selectedLang={VN} />)
      expect(wrap).toMatchSnapshot()
    })
  })
})
