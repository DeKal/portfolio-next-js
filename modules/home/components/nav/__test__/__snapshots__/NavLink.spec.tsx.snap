// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Link Big Device >= 600px should change Language when clicking 1`] = `
<LocalizedComponent
  lang="vn"
  selectedLang="en"
  setActiveLanguage={
    [MockFunction] {
      "calls": Array [
        Array [
          "vn",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  setSelectedLang={
    [MockFunction] {
      "calls": Array [
        Array [
          "vn",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <NavLink
    addTranslation={[Function]}
    addTranslationForLanguage={[Function]}
    ignoreTranslateChildren={false}
    initialize={[Function]}
    lang="vn"
    languages={Array []}
    renderToStaticMarkup={false}
    selectedLang="en"
    setActiveLanguage={
      [MockFunction] {
        "calls": Array [
          Array [
            "vn",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    setSelectedLang={
      [MockFunction] {
        "calls": Array [
          Array [
            "vn",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    translate={[Function]}
  >
    <NavLink__Container
      isSelected={false}
      onClick={[Function]}
    >
      <a
        className="NavLink__Container-sc-13fjz5m-0 fYhrRG"
        onClick={[Function]}
      >
        <Translate
          id="vietnamese"
        >
          <WrappedTranslate
            context={
              Object {
                "activeLanguage": undefined,
                "addTranslation": [Function],
                "addTranslationForLanguage": [Function],
                "defaultLanguage": undefined,
                "ignoreTranslateChildren": false,
                "initialize": [Function],
                "languages": Array [],
                "renderToStaticMarkup": false,
                "setActiveLanguage": [Function],
                "translate": [Function],
              }
            }
            id="vietnamese"
          >
            Missing translationId: vietnamese for language: \${ languageCode }
          </WrappedTranslate>
        </Translate>
      </a>
    </NavLink__Container>
  </NavLink>
</LocalizedComponent>
`;

exports[`Nav Link Big Device >= 600px should render with English selected 1`] = `
<LocalizedComponent
  lang="en"
  selectedLang="en"
>
  <NavLink
    addTranslation={[Function]}
    addTranslationForLanguage={[Function]}
    ignoreTranslateChildren={false}
    initialize={[Function]}
    lang="en"
    languages={Array []}
    renderToStaticMarkup={false}
    selectedLang="en"
    setActiveLanguage={[Function]}
    translate={[Function]}
  >
    <NavLink__Container
      isSelected={true}
      onClick={[Function]}
    >
      <a
        className="NavLink__Container-sc-13fjz5m-0 kOssoX"
        onClick={[Function]}
      >
        <Translate
          id="english"
        >
          <WrappedTranslate
            context={
              Object {
                "activeLanguage": undefined,
                "addTranslation": [Function],
                "addTranslationForLanguage": [Function],
                "defaultLanguage": undefined,
                "ignoreTranslateChildren": false,
                "initialize": [Function],
                "languages": Array [],
                "renderToStaticMarkup": false,
                "setActiveLanguage": [Function],
                "translate": [Function],
              }
            }
            id="english"
          >
            Missing translationId: english for language: \${ languageCode }
          </WrappedTranslate>
        </Translate>
      </a>
    </NavLink__Container>
  </NavLink>
</LocalizedComponent>
`;

exports[`Nav Link Big Device >= 600px should render with Vietnamese selected 1`] = `
<LocalizedComponent
  lang="vn"
  selectedLang="vn"
>
  <NavLink
    addTranslation={[Function]}
    addTranslationForLanguage={[Function]}
    ignoreTranslateChildren={false}
    initialize={[Function]}
    lang="vn"
    languages={Array []}
    renderToStaticMarkup={false}
    selectedLang="vn"
    setActiveLanguage={[Function]}
    translate={[Function]}
  >
    <NavLink__Container
      isSelected={true}
      onClick={[Function]}
    >
      <a
        className="NavLink__Container-sc-13fjz5m-0 kOssoX"
        onClick={[Function]}
      >
        <Translate
          id="vietnamese"
        >
          <WrappedTranslate
            context={
              Object {
                "activeLanguage": undefined,
                "addTranslation": [Function],
                "addTranslationForLanguage": [Function],
                "defaultLanguage": undefined,
                "ignoreTranslateChildren": false,
                "initialize": [Function],
                "languages": Array [],
                "renderToStaticMarkup": false,
                "setActiveLanguage": [Function],
                "translate": [Function],
              }
            }
            id="vietnamese"
          >
            Missing translationId: vietnamese for language: \${ languageCode }
          </WrappedTranslate>
        </Translate>
      </a>
    </NavLink__Container>
  </NavLink>
</LocalizedComponent>
`;

exports[`Nav Link Small Device < 600px should render with English selected 1`] = `
<LocalizedComponent
  lang="en"
  selectedLang="en"
>
  <NavLink
    addTranslation={[Function]}
    addTranslationForLanguage={[Function]}
    ignoreTranslateChildren={false}
    initialize={[Function]}
    lang="en"
    languages={Array []}
    renderToStaticMarkup={false}
    selectedLang="en"
    setActiveLanguage={[Function]}
    translate={[Function]}
  >
    <NavLink__Container
      isSelected={true}
      onClick={[Function]}
    >
      <a
        className="NavLink__Container-sc-13fjz5m-0 kOssoX"
        onClick={[Function]}
      >
        en
      </a>
    </NavLink__Container>
  </NavLink>
</LocalizedComponent>
`;

exports[`Nav Link Small Device < 600px should render with Vietnamese selected 1`] = `
<LocalizedComponent
  lang="vn"
  selectedLang="vn"
>
  <NavLink
    addTranslation={[Function]}
    addTranslationForLanguage={[Function]}
    ignoreTranslateChildren={false}
    initialize={[Function]}
    lang="vn"
    languages={Array []}
    renderToStaticMarkup={false}
    selectedLang="vn"
    setActiveLanguage={[Function]}
    translate={[Function]}
  >
    <NavLink__Container
      isSelected={true}
      onClick={[Function]}
    >
      <a
        className="NavLink__Container-sc-13fjz5m-0 kOssoX"
        onClick={[Function]}
      >
        vn
      </a>
    </NavLink__Container>
  </NavLink>
</LocalizedComponent>
`;
